### PR TITLE
Ports C++ monstermos [dnm pls]

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Hokay.

This is a port of monster860's C++ version of Monstermos (initially made in yogstation13/Yogstation#7981), which is a ZAS-like atmosperics system that runs a *lot* faster than the current system (because it's both done externally and it's just a faster system to begin with) with a lot (and I mean a lot) of gaps in the code filled by beestation's port of the same system as well as additional airlock controllers found at BeeStation/BeeStation-Hornet#1833 (done by ike709.)

Credit to those two aforementioned people, since I barely did anything and I don't deserve any of the credit.

~~**NO MAP CHANGES HAVE BEEN MADE WHATSOEVER, IF YOU TESTMERGE THIS EXPECT THE ENTIRE STATION TO BE ABLE TO BE DEPRESSURIZED *EXTREMELY* QUICKLY IF YOU KNOW WHAT YOU'RE DOING**~~ Maps completely done by BFAT.
## Why It's Good For The Game
it run good, game run good = good game :)

## Changelog
:cl: Mark Suckerberg, BigFatAnimeTiddies, Monster860, ike709
add: Monstermos, a volatile but extremely performant atmospherics system. Don't break any windows!
add: Explosive Decompression is a thing now, and it's TERRIFYING
tweak: Air alarms will turn the lights blue in areas with hazardous atmos.
tweak: Lavaland atmos has a higher minimum pressure.
tweak: Air alarms will turn the lights blue in areas with hazardous atmos.
tweak: Lavaland atmos has a higher minimum pressure.
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
oh yeah yeah

**Comment by Superyodeler** [comment]
I'll take a look at this, it's a big change so it may take some time

**Comment by Superyodeler** [comment]
Been doing some testing, not done yet but it's looking pretty good.

**Comment by Superyodeler** [comment]
Looked at this some more, one problem I see is mapping, as you mentioned. It is possible to cheese a open path from 100 pressure to space without it instantly equalizing by using multiple rooms at different pressures. I have also been able to find tiles with a fairly high difference in pressure. These do equalize gradually, but it looks like it equalizes in a zone based fashion, so probably a feature of ZAS in general. Nothing really game breaking though.
I had been expecting more issues than this, it played fine with atmos tools and the flamethrower worked. @SynnGraffkin may be able to help with some of the maps, and they have said some of the updated ruins from the smartwire revert were taken from Yog which also uses ZAS.

**Comment by SynnGraffkin** [comment]
Tablet atmos analyzers don't work. 
image

**Comment by MarkSuckerberg** [comment]
THIS ERROR WILL CONTINUE TO HAUNT ME FOR THE REST OF MY LIFE

**Comment by SynnGraffkin** [comment]
IT'S THE FUCKING PLAQUES MARK. 

They got updated pathing in TG6 and thus you need to MARG master or the plaques fucking fix the pathing.

**Comment by SynnGraffkin** [comment]
image

**Comment by MarkSuckerberg** [comment]
yepyep, I'm gonna push a potential fix

these plaques will be the death of me

**Comment by triplezeta** [comment]
it keeps happening

**Comment by SynnGraffkin** [comment]
Is that a travis issue now?

File: code/game/machinery/airlock_cycle_control.dm [hunk 1]
```diff
@@ -52,7 +52,6 @@
 	power_channel = AREA_USAGE_ENVIRON
 	req_access = list(ACCESS_ATMOSPHERICS)
 	max_integrity = 250
-	integrity_failure = 80
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 100, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 30)
```
Comment (Maldaris) [inline code/game/machinery/airlock_cycle_control.dm:55]: Probably needs more than just that; Dee suggested armor values be increased, which makes sense.

```suggestion
	armor = list("melee" = 25, "bullet" = 25, "laser" = 25, "energy" = 100, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 30)
```

**Comment by triplezeta** [comment]
Finally

**Pull request merged**